File: internal/model/request/pod.go
package request

type PodCreateRequest struct {
	ChallengeID uint  `binding:"required" json:"challenge_id"`
	TeamID      *uint `json:"team_id"`
	GameID      *uint `json:"game_id"`
	UserID      uint  `json:"-"`
}

type PodFindRequest struct {
	ID          uint  `json:"id" form:"id"`
	ChallengeID uint  `json:"challenge_id" form:"challenge_id"`
	UserID      *uint `json:"user_id" form:"user_id"`
	TeamID      *uint `json:"team_id" form:"team_id"`
	GameID      *uint `json:"game_id" form:"game_id"`
	IsAvailable *bool `json:"is_available" form:"is_available"`
	Page        int   `json:"page" form:"page"`
	Size        int   `json:"size" form:"size"`
}

type PodRemoveRequest struct {
	ID     uint  `json:"-"`
	TeamID *uint `json:"team_id"`
	GameID *uint `json:"game_id"`
	UserID uint  `json:"-"`
}

type PodRenewRequest struct {
	ID     uint  `json:"-"`
	TeamID *uint `json:"team_id"`
	GameID *uint `json:"game_id"`
	UserID uint  `json:"-"`
}

File: web/src/types/pod.ts
import { Game } from "./game";
import { User } from "./user";
import { Team } from "./team";
import { Challenge } from "./challenge";
import { Nat } from "./nat";

export interface Pod {
    id: number;
    name: string;
    game_id: number;
    game: Game;
    user_id: number;
    user: User;
    team_id: number;
    team: Team;
    challenge_id: number;
    challenge: Challenge;
    removed_at: number;
    nats?: Array<Nat>;
}

export interface PodFindRequest {
    id?: number;
    game_id?: number;
    user_id?: number;
    team_id?: number;
    challenge_id?: number;
    is_available?: boolean;
    page?: number;
    size?: number;
}

export interface PodCreateRequest {
    game_id?: number;
    team_id?: number;
    challenge_id?: number;
}

export interface PodRemoveRequest {
    id: number;
    team_id?: number;
    game_id?: number;
}

export interface PodRenewRequest {
    id: number;
    team_id?: number;
    game_id?: number;
}
